products/BellHybrid/apps/application-bell-onboarding/presenter/OnBoardingShortcutsWindowPresenter.hpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include <apps-common/BasePresenter.hpp>
+#include <apps-common/ApplicationCommon.hpp>
+#include <layouts/OnBoardingShortcutsLayouts.hpp>
+
+#include <vector>
+#include <string>
+#include <Item.hpp>
+
+namespace app::OnBoarding
+{
+    class OnBoardingShortcutsWindowContract
+    {
+      public:
+        class View
+        {
+          public:
+            virtual ~View() = default;
+        };
+
+        class Presenter : public BasePresenter<OnBoardingShortcutsWindowContract::View>
+        {
+          public:
+            virtual std::vector<gui::Item *> getLayouts() const      = 0;
+            virtual bool isLastLayout(const gui::Item *layout) const = 0;
+            virtual gui::Item *getFirstLayout() const                = 0;
+        };
+    };
+
+    class OnBoardingShortcutsWindowPresenter : public OnBoardingShortcutsWindowContract::Presenter
+    {
+      private:
+        app::ApplicationCommon *app{};
```

Comment:
I'm wondering, is there a guideline describing how pointers should be initialized?
I'm asking because I'd explicitly write `= nullptr` to increase the readability instead of using value initialization.

Reply:
Furthermore, `app` is set in the constructor so doesn't need to be initialized here. Unless we should always initialize pointers to nullptr, in which case ... (see below).

Reply:
I don't see any guideline describing the initialization of pointers. Furthermore, our code is not consistent. In some places, pointers are always `nullptr` but in other places are without initialization.

If we set the `app` pointer in the constructor then we don't need to explicitly write `nullptr` or `{}`